app/src/middleware/requestSanitizer.ts

```diff
@@ -0,0 +1,50 @@
+import type { NextFunction, Request, Response } from 'express';
+
+/**
+ * Removes all NUL characters from a string
+ * @param str The string to sanitize
+ * @returns The sanitized string if it includes NUL character(s), otherwise the original string
+ */
+function stripNullCharactersFromString(str: string): string {
+  // eslint-disable-next-line no-control-regex
+  return str.replace(/\u0000/g, '');
+}
+
+/**
+ * Recursively sanitize JSON-like data (strings, arrays, plain objects)
+ * @param value The value to sanitize
+ * @returns The sanitized value if a string otherwise the original value
+ */
+function sanitize<T>(value: T): T | string {
```

Comment:
These are utility functions that can fit in many places - may be worth housing them in `/src/utils/utils.ts` or equivalent instead.

Reply:
I was thinking the same but then since this everything coming into our API will hit these middleware I don't see us using these functions anywhere else.